### DISCUSSION TITLE
Close the terminal tab after running a task?

### DISCUSSION THREAD

**Discussion by lucastnr** [zed-industries/zed#12783] [Feature Requests]
I have a setup to open some apps based on the project opened on Zed. 

I've already managed to run it silently by passing the `reveal` as `false`, but every time I open the terminal window there are several terminal tabs from the past tasks I ran.

Is there any way to close the terminal tab after running a task, maybe a configuration I'm not aware of?

![image](https://github.com/zed-industries/zed/assets/59018759/fbd85ea2-7b1a-490a-b1cb-541908406d02)

I'll leave my `tasks.json` and `keymap.json` files below

```jsonc
// tasks.json file
[
  {
    "label": "Open github",
    "command": "open \"x-github-client://openRepo/$(git config --get remote.origin.url)\"",
    "use_new_terminal": true,
    "allow_concurrent_runs": false,
    "reveal": "never"
  },
  {
    "label": "Open warp",
    "command": "open \"warp://action/new_tab?path=$ZED_WORKTREE_ROOT/\"",
    "use_new_terminal": true,
    "allow_concurrent_runs": false,
    "reveal": "never"
  }
]
```

```jsonc
// keymaps.json file
// ...
  {
    "bindings": {
      "cmd-shift-g": [
        "task::Spawn",
        {
          "task_name": "Open github"
        }
      ]
    }
  },
  {
    "bindings": {
      "cmd-shift-c": [
        "task::Spawn",
        {
          "task_name": "Open warp"
        }
      ]
    }
  },
// ...
```

**Comment by christophevg** [comment]
I think `"hide": "always"` (or "on success" if you like) might be what you were looking for?

See: [https://zed.dev/docs/tasks](https://zed.dev/docs/tasks) 
```
    // What to do with the terminal pane and tab, after the command has finished:
    // * `never` — Do nothing when the command finishes (default)
    // * `always` — always hide the terminal tab, hide the pane also if it was the last tab in it
    // * `on_success` — hide the terminal tab on task success only, otherwise behaves similar to `always`
    "hide": "never",
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
